content/software-triage.md

```diff
@@ -0,0 +1,311 @@
+- Out of date OS libraries.
+    Many applications refer to system libraries. Some programs may be expecting to find a specific version of a library.
+- Out of date drivers.
+    Older kernels may be using out of date driver software.
+- Incorrect Drivers.
+    If an unneeded or incorrect driver is installed this can cause unexpected side effects.
+- Kernel Bugs
+
+## Solutions
+
+We troubleshoot issues with the approach of taking small steps, then gradually larger steps:
+
+ 0. Backups - Preserve
+ 1. Get logs - Observe
+
+    **Non Invasive**
+
+ 2. Check for updates.
+ 3. Fix the package manager.
+ 4. Update with packages held.
+ 5. Test 2nd Admin User.
+ 6. Test in a Live Disk
+
+    **Surgical - Invasive but Reversible**
+
+ 7. Remove or install individual packages.- Surgically Invasive, easy to reverse.
+ 8. Try a Different Kernel|-- System modifications, global changes. Invasive.
+
+     **Global - Invasive and Irreversible**
+
+ 9. Upgrade OS
+ 10. Reinstall the OS.
+
+### 1. Get Logs
+
+To generate the logs we need, please click on **Activities** in the top left of your screen, search for System76, and then click on the System76 Driver application.
+
+Then click on the **Create Log Files** button and the log file will be in your home directory when you first open the Files application.
+
+For more information, please see this support article: [Creating Log Files](/articles/log-files)
+
+### 2. Update the Installed OS
+
+Make sure the version of the OS that is installed is fully up to date.
+
+NOTE: The following instructions can be used in either Ubuntu or Pop!\_OS.
+
+You can open the terminal by clicking Activities in the upper-left corner and searching for "Terminal", after pressing <kbd>SUPER</kbd>.
+When you run the first command with "sudo" it will ask you to enter your password.
+Typing your password will not show anything in the Terminal window. That is normal and for security.
+Just enter your password as you normally would, and then hit <kbd>ENTER</kbd>. You'll also need to hit <kbd>ENTER</kbd> after each command.
+
+```bash
+sudo apt update
+sudo apt full-upgrade
+reboot
+```
+
+### 3. Fix Package Manager Issues
+
+```bash
+sudo apt clean
+sudo apt update -m
+sudo dpkg --configure -a
+sudo apt install -f
+sudo apt dist-upgrade
+sudo apt autoremove --purge
+reboot
+```
+
+<!--> May want to switch this to the version with extra steps and built in log collecting. -->
+
+Source: [Fix Package Manager](/articles/package-manager-pop)
+
+### 4. Update with Packages Held
+
+To perform an update while keeping some packages on their current version, use the following Terminal command:
+
+```bash
+sudo apt-mark hold [packagename]
+```
+
+If the package is not central to OS functions, it can be held at the current version and the other packages updated around it. If the problem doesn't present itself when a package has been held, it was likely being updated to a newer version that was causing instability.
```

Comment:
I would caution against including holding a package this high up in the troubleshooting process.
- For one thing, you haven't explained how to downgrade a package yet, so if someone's having a problem and they follow this step, they would be holding the version that was causing the problem.
  - You also haven't explained how to identify which package to downgrade/hold.
- Also, holding packages is almost never a permanent solution. It should usually only be a temporary workaround. (If it's working around a bug, the bug needs to be reported/fixed; if it's working around a workflow change, the user needs to adapt their workflow to the new version before their old version stops working.)

Reply:
I agree with this. The use of holding packages could be made clearer. I'll work your feedback into a revision.